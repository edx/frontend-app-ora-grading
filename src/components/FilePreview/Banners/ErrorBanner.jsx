import React from 'react';
import PropTypes from 'prop-types';

import { Alert, Button } from '@edx/paragon';
import { Info } from '@edx/paragon/icons';
import { FormattedMessage } from '@edx/frontend-platform/i18n';

const messageShape = PropTypes.shape({
  id: PropTypes.string,
  defaultMessage: PropTypes.string,
});

export const ErrorBanner = ({ actions, headerMessage, children }) => {
  const actionButtons = actions.map(action => (
    <Button key={action.id} onClick={action.onClick} variant="outline-primary">
      <FormattedMessage {...action.message} />
    </Button>
  ));
  return (
    <Alert variant="danger" icon={Info} actions={actionButtons}>
      <Alert.Heading>
        <FormattedMessage {...headerMessage} />
      </Alert.Heading>
      {children}
    </Alert>
  );
};
ErrorBanner.defaultProps = {
  actions: [],
  children: null,
};
ErrorBanner.propTypes = {
  actions: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      onClick: PropTypes.func,
      message: messageShape,
    }),
  ),
  headerMessage: messageShape.isRequired,
  children: PropTypes.node,
};

export default ErrorBanner;
