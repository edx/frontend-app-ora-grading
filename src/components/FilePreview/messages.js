import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  fileInfo: {
    id: 'ora-grading.InfoPopover.fileInfo',
    defaultMessage: 'File info',
    description: 'Popover trigger button text for file preview card',
  },
});

export default messages;
