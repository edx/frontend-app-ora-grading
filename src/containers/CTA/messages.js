/* eslint-disable quotes */
import { defineMessages } from '@edx/frontend-platform/i18n';
import { StrictDict } from 'utils';

const messages = defineMessages({
  ctaFeedbackMessage: {
    id: 'ora-grading.CTA.feedbackMessage',
    defaultMessage: 'Thanks for using the new ORA staff grading experience. We\'d love to hear your feedback',
    description: 'Thank user for using ora and ask for feed back',
  },
  ctaLinkMessage: {
    id: 'ora-grading.CTA.linkMessage',
    defaultMessage: 'here',
    description: 'placeholder for the feedback anchor link',
  },
});

export default StrictDict(messages);
