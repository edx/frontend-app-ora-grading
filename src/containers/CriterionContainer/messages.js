import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  addComments: {
    id: 'ora-grading.CriterionFeedback.addCommentsLabel',
    defaultMessage: 'Add comments',
    description: 'label for editable feedback field',
  },
  comments: {
    id: 'ora-grading.CriterionFeedback.commentsLabel',
    defaultMessage: 'Comments',
    description: 'label for read-only feedback field',
  },
  optionPoints: {
    id: 'ora-grading.RadioCriterion.optionPoints',
    defaultMessage: '{points} points',
    description: 'criterion option point value display',
  },
});

export default messages;
