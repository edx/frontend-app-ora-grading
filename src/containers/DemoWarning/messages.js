/* eslint-disable quotes */
import { defineMessages } from '@edx/frontend-platform/i18n';
import { StrictDict } from 'utils';

const messages = defineMessages({
  demoModeHeading: {
    id: 'ora-grading.ReviewModal.demoHeading',
    defaultMessage: 'Demo Mode',
    description: 'Demo mode heading',
  },
  demoModeMessage: {
    id: 'ora-grading.ReviewModal.demoMessage',
    defaultMessage: 'You are using the Demo Mode of the new Enhanced ORA Staff Grader interface.  You will be unable to submit grades until you activate the feature.',
    description: 'Demo mode message',
  },
});

export default StrictDict(messages);
