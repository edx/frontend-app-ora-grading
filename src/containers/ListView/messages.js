import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  backToResponses: {
    id: 'ora-grading.ListView.ListViewBreadcrumbs.backToResponses',
    defaultMessage: 'Back to all open responses',
    description: 'Breadcrumbs link text to return to ORA list in LMS.',
  },
  noResultsFound: {
    id: 'ora-grading.ListView.noResultsFound',
    defaultMessage: 'No results found',
    description: 'Empty table content for submissions list',
  },
  viewAllResponses: {
    id: 'ora-grading.ListView.viewAllResponses',
    defaultMessage: 'View all responses',
    description: 'Button text to load all responses for review/grading',
  },
  viewSelectedResponses: {
    id: 'ora-grading.ListView.viewSelectedResponses',
    defaultMessage: 'View selected responses ({value})',
    description: 'Button text to load selected responses for review/grading',
  },
  username: {
    id: 'ora-grading.ListView.tableHeaders.username',
    defaultMessage: 'Username',
    description: 'Username table column header for submission list view',
  },
  teamName: {
    id: 'ora-grading.ListView.tableHeaders.teamName',
    defaultMessage: 'Team name',
    description: 'Team name table column header for submission list view',
  },
  learnerSubmissionDate: {
    id: 'ora-grading.ListView.tableHeaders.learnerSubmissionDate',
    defaultMessage: 'Learner submission date',
    description: 'Learner submission date table column header for submission list view',
  },
  teamSubmissionDate: {
    id: 'ora-grading.ListView.tableHeaders.teamSubmissionDate',
    defaultMessage: 'Team submission date',
    description: 'Team submission date table column header for submission list view',
  },
  grade: {
    id: 'ora-grading.ListView.tableHeaders.grade',
    defaultMessage: 'Grade',
    description: 'Grade table column header for submission list view',
  },
  gradingStatus: {
    id: 'ora-grading.ListView.tableHeaders.gradingStatus',
    defaultMessage: 'Grading status',
    description: 'Grading status table column header for submission list view',
  },
});

export default messages;
