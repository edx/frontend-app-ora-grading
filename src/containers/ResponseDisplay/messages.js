import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  tableNameHeader: {
    id: 'ora-grading.ResponseDisplay.SubmissionFiles.tableNameHeader',
    defaultMessage: 'Name',
    description: 'Table header file name',
  },
  tableExtensionHeader: {
    id: 'ora-grading.ResponseDisplay.SubmissionFiles.tableExtensionHeader',
    defaultMessage: 'File Extension',
    description: 'Table header file extension',
  },
  tablePopoverHeader: {
    id: 'ora-grading.ResponseDisplay.SubmissionFiles.tablePopoverHeader',
    defaultMessage: 'File Metadata',
    description: 'Table header for popover file metadata',
  },
});

export default messages;
