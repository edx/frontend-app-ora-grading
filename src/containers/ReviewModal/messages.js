import { defineMessages } from '@edx/frontend-platform/i18n';
import { StrictDict } from 'utils';

const messages = defineMessages({
  reloadSubmission: {
    id: 'ora-grading.ReviewModal.reloadSubmission',
    defaultMessage: 'Reload submission',
    description: 'Reload button text in case of network failure',
  },
  loadingResponse: {
    id: 'ora-grading.ReviewModal.loadingResponse',
    defaultMessage: 'Loading response',
    description: 'loading text for submission response review screen',
  },
  gradeNotSubmittedHeading: {
    id: 'ora-grading.ReviewModal.gradeNotSubmitted.heading',
    defaultMessage: 'Grade not submitted',
    description: 'Grade submission network error heading',
  },
  gradeNotSubmittedContent: {
    id: 'ora-grading.ReviewModal.gradeNotSubmitted.heading',
    defaultMessage: "We're sorry, something went wrong when we tried to submit this grade.  Please try again.",
    description: 'Grade submission network error message',
  },
  resubmitGrade: {
    id: 'ora-grading.ReviewModal.resubmitGrade',
    defaultMessage: 'Resubmit grate',
    description: 'Resubmit grade button after network failure',
  },
  dismiss: {
    id: 'ora-grading.ReviewModal.dismiss',
    defaultMessage: 'Dismiss',
    description: 'Dismiss error action button text',
  },
});

export default StrictDict(messages);
