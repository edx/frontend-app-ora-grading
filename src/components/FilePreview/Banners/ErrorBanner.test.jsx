import React from 'react';
import { shallow } from 'enzyme';

import ErrorBanner from './ErrorBanner';

import messages from '../messages';

describe('Error Banner component', () => {
  const children = <p>Abitary Child</p>;

  const props = {
    actions: [
      {
        id: 'action1',
        onClick: jest.fn().mockName('action1.onClick'),
        message: messages.retryButton,
      },
      {
        id: 'action2',
        onClick: jest.fn().mockName('action2.onClick'),
        message: messages.retryButton,
      },
    ],
    headingMessage: messages.unknownError,
    children,
  };

  let el;
  beforeEach(() => {
    el = shallow(<ErrorBanner {...props} />);
  });

  test('snapshot', () => {
    expect(el).toMatchSnapshot();
  });

  describe('component', () => {
    test('children node', () => {
      expect(el.containsMatchingElement(children)).toEqual(true);
    });

    test('verify actions', () => {
      const actions = el.find('Alert').prop('actions');
      expect(actions).toHaveLength(props.actions.length);

      actions.forEach((action, index) => {
        expect(action.type).toEqual('Button');
        expect(action.props.onClick).toEqual(props.actions[index].onClick);
        // action message
        expect(action.props.children.props).toEqual(props.actions[index].message);
      });
    });

    test('verify heading', () => {
      const heading = el.find('FormattedMessage');
      expect(heading.props()).toEqual(props.headingMessage);
    });
  });
});
