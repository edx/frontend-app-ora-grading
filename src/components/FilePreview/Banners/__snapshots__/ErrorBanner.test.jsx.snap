// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Banner component snapshot 1`] = `undefined`;
